src/main/ruby/core/kernel.rb

```diff
@@ -464,6 +464,14 @@ def tap
     self
   end
 
+  def yield_self
+    if block_given?
+      yield self
+    else
+      Enumerator.new(1) { |yielder| yielder.yield self }
```

Comment:
There is a slightly simpler pattern for enumerator, for example look at:
https://github.com/graalvm/truffleruby/blob/6501b86f9f175465bfd2f943a5dd9334e9136729/src/main/ruby/core/enumerable.rb#L175